azext_iot/digitaltwins/providers/import_job.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azext_iot.common.utility import handle_service_exception
+from azext_iot.digitaltwins.providers.base import DigitalTwinsProvider
+from azext_iot.digitaltwins.providers import ErrorResponseException
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azure.cli.core.azclierror import CLIInternalError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class ImportJobProvider(DigitalTwinsProvider):
+    def __init__(self, cmd, name, rg=None):
+        super(ImportJobProvider, self).__init__(cmd=cmd, name=name, rg=rg)
+        self.sdk = self.get_sdk().import_jobs
+        self.cli = EmbeddedCLI()
+
+    def _get_blob_url(self, blob_name, blob_container, storage_account):
+        storage_account_cstring_op = self.cli.invoke(
+            "storage account show-connection-string -n '{}'".format(storage_account)
+        )
+        if not storage_account_cstring_op.success():
+            raise CLIInternalError(
```

Comment:
make this (and all other CLIInternalErrors) the more specific azure error or Ryan will be sad... 

This specific example looks like it would benefit from         handle_service_exception(e) 

look in azure.cli.core.azclierror to see more examples.

Reply:
Changes to the more specific '''ResourceNotFoundError'''